mimic/test/test_ironic.py

```diff
@@ -0,0 +1,247 @@
+import json
+from uuid import uuid4
+
+from twisted.trial.unittest import SynchronousTestCase
+from twisted.internet.task import Clock
+
+from mimic.core import MimicCore
+from mimic.resource import MimicRoot
+from mimic.test.helpers import json_request, request
+from mimic.model.ironic_objects import IronicNodeStore, Node
+
+
+class IronicNodeStoreUnitTests(SynchronousTestCase):
+    """
+    Unit test for `node_by_id` in :obj:`IronicNodeStore`
+    """
+
+    def test_node_by_id(self):
+        """
+        Test for :func:`node_by_id`
+        """
+        node_list = [Node(node_id=str(uuid4())) for each in range(2)]
+        node_store = IronicNodeStore(ironic_node_store=node_list)
+        self.assertFalse(node_store.node_by_id('111'))
+
+
+class IronicAPITests(SynchronousTestCase):
+
+    """
+    Tests for the Ironic api
+    """
+
+    def setUp(self):
+        """
+        Initialize core and root
+        """
+        self.core = MimicCore(Clock(), [])
+        self.root = MimicRoot(self.core).app.resource()
+        self.node_details_attributes = [
+            "instance_uuid", "target_power_state",
+            "properties", "uuid", "driver_info", "target_provision_state",
+            "last_error", "console_enabled", "extra", "driver", "links",
+            "maintenance_reason", "updated_at", "provision_updated_at",
+            "maintenance", "provision_state", "reservation", "created_at",
+            "power_state", "instance_info", "ports"
+        ]
+
+    def get_nodes(self, postfix=None):
+        """
+        Get nodes and return content
+        """
+        url = "/ironic/v1/nodes"
+        if postfix:
+            url = "/ironic/v1/nodes" + postfix
+        (response, content) = self.successResultOf(json_request(
+            self, self.root, "GET", url))
+        self.assertEqual(200, response.code)
+        return content
+
+    def test_list_nodes(self):
+        """
+        Test ``/nodes`` to return response code 200 and validate the
+        attributes of the json response body
+        """
+        content = self.get_nodes()
+        for each_node in content['nodes']:
+            self.assertEqual(
+                sorted(each_node.keys()),
+                sorted(['instance_uuid', 'uuid', 'links', 'maintenance',
+                        'provision_state', 'power_state']))
+
+    def test_list_nodes_with_details(self):
+        """
+        Test ``/nodes/detail`` to return response code 200 and validate a
+        response body exists.
+        """
+        content = self.get_nodes('/detail')
+        self.assertTrue(len(content['nodes']) > 1)
```

Comment:
@jimrollenhagen : Currently, list nodes is set to always return a list of 11 nodes, I plan on changing this to be configurable using the control plane someday. 
Also, this will change once I have added the add and delete node APIs.